Rocket.Chat/Controllers/Chat/ChatViewController.swift

```diff
@@ -116,6 +120,21 @@ final class ChatViewController: SLKTextViewController {
         }
     }
 
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        let menuItemFilter: (UIBarButtonItem) -> Bool = { barButtonItem in
+            if let button = barButtonItem.customView as? UIButton,
+                let sideMenuController = self.sideMenuController {
+                return button.allTargets.contains(sideMenuController)
+            }
+            return false
+        }
+        guard let leftMenuButtonItem = self.navigationItem.leftBarButtonItems?.filter(menuItemFilter).first else { return }
+
+        leftMenuButtonItem.addBadge(self.menuBadge)
```

Comment:
I understand what you're doing here, but I've another idea, let me know what you think.

1. Create a protocol of UIBarButtonItem that contains the badge.
2. Create a new button following this protocol and having the badge.
3. Remove the default button from `SideMenuController.preferences.drawing.menuButtonImage` in `MainChatViewController.swift`
4. Use the new button created, an calling the methods related to the badge that follows the protocol.

What do you think?